enhancements/ingress/route-secret-injection-for-external-certificate-management.md

```diff
@@ -0,0 +1,390 @@
+  - Verify private key matches public certificate
+
+- Validations done by API server as part of [ValidateRoute()](https://github.com/openshift/openshift-apiserver/blob/aac3dd5bf0547e928103a0f718ca104b1bb13930/pkg/route/apis/route/validation/validation.go#L21),
+
+  - The router serviceaccount must have permission to read this secret particular secret.
+    - The role and rolebinding to provide this access must be provided by the user.
+  - CEL validations and o/library-go will enforce that both `.spec.tls.certificate` and `.spec.tls.externalCertificate`
+    are not specified on the route.
+
+The router being an edge component, from a security standpoint is more prone to
+being compromised. In order to avoid providing the router with escalated privileges
+to read all secrets, the router will implement a single item list/watch for secrets (secret monitor).
+This uses name-scoped rbac (created by the user) to access the particular secrets.
+
+A watch based secret monitor will be introduced in the router in order to keep
+track of all the secrets referenced by the routes. This component is solely
+responsible for maintaining all the single item list-watch functions required
+to cache the referenced secrets.
+
+The router will bootstrap the secret monitor to ensure it can keep the
+certificate/secret up-to-date. This means the router pod will maintain active watches
+for every secret that is referenced by a route.
+
+Every active watch will be linked to a route, meaning the secret monitor
+will be linked to the lifecycle of the route. For every new route that is created,
+the secret monitor will start a watch if the route uses `.spec.tls.externalCertificate`.
+If a route is deleted, the secret monitor will deregister the route and teardown
+the watch associated with it.
+
+The `ServiceAliasConfig` creation logic will be updated in the router to also parse
+the secret referenced in `.spec.tls.externalCertificate`. The router will
+use the default certificates only when `.spec.tls.certificate/key` or `.spec.tls.externalCertificate`
+are not provided.
+
+The cluster-ingress-operator will propagate the relevant Tech-Preview feature gate down to the
+router. This feature gate will be added as a command-line argument called `ROUTER_EXTERNAL_CERTIFICATE`
+to the router and will not be user configurable.
+
+### Risks and Mitigations
+
+There is a possibility of an invalid route being processed by the router (edge case),
+if any changes are done to the rbac or the referenced secret is deleted after the API
+server validation but before router has processed the route (maybe router pod is not running)
+then this can lead to the router processing this incorrect route.
+
+> Will need to duplicate the validations present on the API server to the router.
+
+### Drawbacks
+
+The user will need to manually create, provide and maintain the rbac required by the
+router so that it can access secrets securely. This becomes tedious when users have
+many of Routes.
```

Comment:
```suggestion
The user will need to manually create, provide and maintain the rbac required by the
router so that it can access secrets. This becomes tedious when users have
many routes.
```